src/XMakeBuildEngine/Logging/BaseConsoleLogger.cs

```diff
@@ -1020,7 +1020,9 @@ public virtual void Initialize(IEventSource eventSource)
 
             if (showOnlyWarnings || showOnlyErrors)
             {
-                _showSummary = false;
+                if (!_showSummary.HasValue)
+                    _showSummary = false;
```

Comment:
Add `{ }` here and a comment that we default to false if the user didn't specify `Summary`